src/TorchVision/dsets/CelebA.cs

```diff
@@ -0,0 +1,276 @@
+                } else {
+                    headers = null;
+                }
+
+                var indices = data.Select(row => row[0]).ToArray();
+
+                data = data.Select(row => row.AsSpan(1).ToArray()).ToList();
+                var data_long = new long[data.Count, data[0].Length];
+                for (int i = 0; i < data_long.GetLength(0); i++) {
+                    for (int j = 0; j < data_long.GetLength(1); j++) {
+                        if (data[i].Length != data_long.GetLength(1)) {
+                            throw new Exception();
+                        }
+                        long result = long.TryParse(data[i][j], out result) ? result : 0;
+                        data_long[i, j] = result;
+                    }
+                }
+                return new CSV(headers, indices, torch.tensor(data_long));
+            }
+
+            /// <summary>
+            /// Get tensor according to index
+            /// </summary>
+            /// <param name="index">Index for tensor</param>
+            /// <returns>Tensors of index.</returns>
+            public override Dictionary<string, Tensor> GetTensor(long index)
+            {
+                if (this.filename is null) throw new InvalidOperationException();
+                Tensor X = torchvision.io.read_image(Path.Combine(this.root, base_folder, "img_align_celeba", this.filename[index]));
+
+                var result = new Dictionary<string, Tensor>();
+                foreach (var t in this.target_type) {
+                    if (t == "attr") {
+                        if (this.attr is null) throw new InvalidDataException();
+                        result[t] = this.attr[index, TensorIndex.Colon];
+                    } else if (t == "identity") {
+                        if (this.identity is null) throw new InvalidDataException();
+                        result[t] = this.identity[index, 0];
+                    } else if (t == "bbox") {
+                        if (this.bbox is null) throw new InvalidDataException();
+                        result[t] = this.bbox[index, TensorIndex.Colon];
+                    } else if (t == "landmarks") {
+                        if (this.landmarks_align is null) throw new InvalidDataException();
+                        result[t] = this.landmarks_align[index, TensorIndex.Colon];
+                    } else {
+                        throw new InvalidDataException($"Target type \"{t}\" is not recognized.");
+                    }
+                }
+
+                if (this.transform is not null) {
+                    X = this.transform.call(X);
+                }
+                result["input"] = X;
+                if (this.target_transform is not null) {
+                    string t = this.target_type[0];
+                    result[t] = this.target_transform.call(result[t]);
+                }
+                return result;
+            }
+
+            private bool _check_integrity()
```

Comment:
For non-public APIs, and APIs that are not copies of PyTorch APIs, we usually try to stay with .NET naming conventions. I'm not sure that we've done it consistently, but it's been the intent.

Reply:
Thanks. `CelebA._check_integrity()` is a non-public API but a copy of a PyTorch API (https://github.com/pytorch/vision/blob/b78d98bb152ffb9c0c0f5365f59f475c70b1784e/torchvision/datasets/celeba.py#LL132C9-L132C25). Keeping the same name makes it easier to track the changes in the original PyTorch code. Should we still stay with .NET naming conventions in this case?

Reply:
Fixed by 377a802da22b963dc248b8248b3244ed2e6deadb